core/client/fs/src/main/java/alluxio/client/file/FileSystemContext.java

```diff
@@ -284,19 +241,15 @@ public synchronized void close() throws IOException {
         pool.close();
       }
       mBlockWorkerClientPool.clear();
-
-      if (mMetricsMasterClient != null) {
-        ThreadUtils.shutdownAndAwaitTermination(mExecutorService,
-            mClientContext.getConf().getMs(PropertyKey.METRICS_CONTEXT_SHUTDOWN_TIMEOUT));
-        mMetricsMasterClient.close();
-        mMetricsMasterClient = null;
-        mClientMasterSync = null;
+      if (mClientContext.getConf().getBoolean(PropertyKey.USER_METRICS_COLLECTION_ENABLED)) {
```

Comment:
Essentially, yes - technically, no. Upon creation, it is likely that the cluster defaults have not been loaded. Pulling in the cluster defaults could change this property. Other than that change, it is immutable.

I think I would prefer to save it to a variable, and read that value. In this case, cluster metrics can only be enabled by client-side configuration, not cluster defaults. Is that behavior we would like to have?

Reply:
I think it would be quite valuable to be able to enable/disable cluster defaults from the master. Could we make sure cluster defaults have been downloaded before checking the property?

Reply:
It wouldn't really fit with the design of the current implementation because we load cluster defaults lazily upon first connection. Typically, that means when the user application opens or reads a file. @cheng-chang was also having issues with cluster defaults being loaded lazily as it was affecting path configuration. We may be changing it in the near future.

Personally, I don't think the effort is worth changing right now. If we decide to pro-actively load cluster defaults, then supporting enabling metrics via cluster defaults should become trivial.

Reply:
ok, we can let the dust settle on Cheng's changes before putting work in here